### PR TITLE
Add remaining Longest Road functionality

### PR DESCRIPTION
**Pull request by abellagonzalo**
Init Longest Road State when building the board.
Update Longest Road State if needed after building a Settlement

### CODE REVIEW

File: contracts/src/main/kotlin/com/contractsAndStates/contracts/LongestRoadContract.kt [hunk 1]
```diff
@@ -16,16 +16,33 @@ class LongestRoadContract : Contract {
...
+                 */
+
+                "Longest Road cannot be assigned to any player at the beggining of the game" using (
+                        outputLongestRoadState.holder == null)
+
+                /**
+                 *  ******** SIGNATURES ********
+                 */
+
+                val signingParties = tx.commandsOfType<Commands.Init>().single().signers.toSet()
```
Comment (xavierlepretre) [inline contracts/src/main/kotlin/com/contractsAndStates/contracts/LongestRoadContract.kt:34]: `command.value.signers.toSet()`?

File: contracts/src/main/kotlin/com/contractsAndStates/contracts/LongestRoadContract.kt [hunk 2]
```diff
@@ -16,16 +16,33 @@ class LongestRoadContract : Contract {
...
+                val participants = outputLongestRoadState.participants.map{ it.owningKey }
+
+                "All players must verify and sign the transaction to build a settlement." using(
+                        signingParties.containsAll<PublicKey>(participants) && signingParties.size == 4)
```
Comment (xavierlepretre) [inline contracts/src/main/kotlin/com/contractsAndStates/contracts/LongestRoadContract.kt:38]: Not sure we should hardcode the number of participants here: 4.

File: workflows/src/main/kotlin/com/flows/BuildSettlementFlow.kt [hunk 3]
```diff
@@ -43,36 +45,58 @@ class BuildSettlementFlow(
...
+        // Step 12. Add all states and commands to the transaction.
         tb.addInputState(gameBoardStateAndRef)
         tb.addReferenceState(turnTrackerReferenceStateAndRef)
         tb.addOutputState(settlementState, BuildPhaseContract.ID)
         tb.addOutputState(newBoardBuilder.build())
         tb.addCommand(GameStateContract.Commands.UpdateWithSettlement(), gameBoardState.players.map { it.owningKey })
+        if (outputLongestRoadState.holder != longestRoadState.holder) {
+            tb.addInputState(longestRoadStateStateAndRef)
+            tb.addOutputState(outputLongestRoadState, LongestRoadContract.ID)
+        }
```
Comment (xavierlepretre) [inline workflows/src/main/kotlin/com/flows/BuildSettlementFlow.kt:99]: The `Claim` command should be added too, no?

Comment (xavierlepretre) [reply workflows/src/main/kotlin/com/flows/BuildSettlementFlow.kt:99]: @abellagonzalo Bump^

**Comment by xavierlepretre** [comment]
Where did all this go? I cannot see Gonzalo's work in the file itself: https://github.com/rogersanick/SettlersOfCordan/blob/master/workflows/src/main/kotlin/com/flows/BuildSettlementFlow.kt

**Comment by xavierlepretre** [comment]
It is not visible here https://github.com/rogersanick/SettlersOfCordan/commits/master

**Comment by rogersanick** [comment]
@xavierlepretre - I merged the copy locally and then pushed to master but it was crediting me for all of the changes.

I have since rolled back and asked Gonzalo to give me access to his fork so that I can push my updates.

**Comment by rogersanick** [comment]
And then make a new pr*

**Pull request merged**